azext_iot/central/providers/device_group_provider.py

```diff
@@ -52,3 +56,92 @@ def list_device_groups(
         )
 
         return device_groups
+
+    def get_device_group(
+        self, device_group_id, central_dns_suffix=CENTRAL_ENDPOINT
+    ) -> Union[DeviceGroupPreview, DeviceGroupV1_1_preview]:
+        # get or add to cache
+        device_group = self._device_groups.get(device_group_id)
+        if not device_group:
+            device_group = central_services.device_group.get_device_group(
```

Comment:
will this really not just throw an error if you provide a bad device group (non existant)? what if you provide the wrong app id?

Reply:
I followed the pattern in `get_device_template` and I think it will raise `AzureResponseError` error in service utility in `try_extract_result` function if the response is 404.

Reply:
Yes, I saw those services using try-except to handle exceptions in a generic handling function. But I think my way here should work as well.

Reply:
Services throw out error messages as they come out from API call. Almost all central commands are wrappers around the public APIs so I would keep it as is and let throw at service level.

Reply:
@vilit1 Remove old preview versions shouldn't have big user impact and it'll make codebase cleaner.

Reply:
Actually we support all versions in this cli extension right now: preview, 1.1-preview and v1(GA version). One more preview version: 1.2-preview doesn't exist here. 😕 
I'm going to add 2022-05-31(GA) here. And we also have one new preview version and one new GA version incoming.
Since we're going to retire preview and 1.1-preview, we want to only keep one latest preview version in cli extension repo. Users still can utilize previous CLI release version if they want to use old API.
@yang-hai-feng

Reply:
make sure it is clearly documented which extension versions support which preview versions. Would be good to have this in a read me somewhere (as a table format of when we started to support the version to when we stopped)

Reply:
Sounds good. We'll do a  refactor in iot central cli extension to use SDK but not directly api call after this PR and we'll remove the old version in that one. We'll keep only latest preview and GA version in the future.